File: 1009-complement-of-base-10-integer/1009-complement-of-base-10-integer.py
class Solution:
    def bitwiseComplement(self, n: int) -> int:
        if n == 0:
            return 1
        mask = 1 << 31
        while mask > 0 and  not (mask & n):
            mask >>= 1
        while mask > 0:
            n ^= mask
            mask >>= 1
        return n

File: 1009-complement-of-base-10-integer/1009-complement-of-base-10-integer.cpp
class Solution {
public:
    int bitwiseComplement(int n) {
        if(n == 0)
        {
            return 1;
        }
        unsigned int mask = 1 << 31;
        while(mask > 0 && !(mask & n))
        {
            mask >>= 1;
        }
        while(mask > 0)
        {
            n ^= mask;
            mask >>= 1;
        }
        return n;
    }
};